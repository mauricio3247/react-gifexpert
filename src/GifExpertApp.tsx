import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGridComponent } from './components/GifGrid';

type GifExperProps= {
    defaultCategories?: string[]
}

export const GifExperApp = ({defaultCategories = [] }: GifExperProps )=> {
    
    const [categories, setCategories] = useState<string[]>(defaultCategories)

    return (
        <>
            <h1>GifExpertApp</h1>
            <hr/>

            <AddCategory setCategories = {setCategories} />
            <ol>
                { 
                categories.map ((item) => (
                    <GifGridComponent category={item} key={item}></GifGridComponent>
                ))
                }
            </ol>
        </>
    )
}