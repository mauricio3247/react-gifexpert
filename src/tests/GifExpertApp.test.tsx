import React from 'react'
import { shallow } from 'enzyme'
import { GifExperApp } from '../GifExpertApp'

describe('test componente GifExpertApp', () => {

    test('Debe crearse el componente', ()=> {

        const wrapper = shallow (<GifExperApp />)
        expect (wrapper).toMatchSnapshot()
    })

    test('Debe crearse el componente', ()=> {
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow (<GifExperApp defaultCategories={categories} />)
        expect (wrapper).toMatchSnapshot() 
        expect (wrapper.find('GifGridComponent').length).toBe(categories.length)
    })



})