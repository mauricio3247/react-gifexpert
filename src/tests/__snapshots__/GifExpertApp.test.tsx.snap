// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test componente GifExpertApp Debe crearse el componente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`test componente GifExpertApp Debe crearse el componente 2`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGridComponent
      category="One Punch"
      key="One Punch"
    />
    <GifGridComponent
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
